### PR TITLE
[iOS][Circle CI] Add PR jobs for iOS builds

### PR DESCRIPTION
**Pull request by xta0**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25840 [iOS][Circle CI] Add PR jobs for iOS builds**

### Summary:

The CI jobs for iOS builds are missing, this PR creates two jobs in Cricle CI's PR workflow

- pytorch_ios_10_2_1_x86_64_build
- pytorch_ios_10_2_1_arm64_build

### Note:

Those two jobs will not store any artifacts nor upload any binary files, which will be done in the next PR.

Test Plan:

- Jobs can be triggered by any PR.
- Jobs can be run successfully.

Differential Revision: [D17255504](https://our.internmc.facebook.com/intern/diff/D17255504)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,7 +63,17 @@ macos_brew_update: &macos_brew_update
...
+  name: Brew update and install iOS toolchains
+  no_output_timeout: "1h"
+  command: |
+    set -ex
+    brew update
+    brew unlink parallel
+    brew install moreutils
+    brew link parallel --overwrite
+    brew install expect
+    brew install libtool
```
Comment (ezyang) [inline .circleci/config.yml]: This seems like a direct copy paste `binary_macos_brew_update` (except that it is missing the `brew untap caskroom/homebrew-cask` command, but *arguably* it should do that here too). Can't we just use `binary_macos_brew_update` instead (perhaps renaming it to more clearly say something like, `macos_brew_update_nolibomp`)? Without deduplicating, there's now **three** places you'd have to update if there were tweaks to this block.

File: .circleci/verbatim-sources/pytorch-build-params.yml [hunk 2]
```diff
@@ -18,3 +18,23 @@ pytorch_params: &pytorch_params
     USE_CUDA_DOCKER_RUNTIME: << parameters.use_cuda_docker_runtime >>
   resource_class: << parameters.resource_class >>
 
+pytorch_ios_params: &pytorch_ios_params
```
Comment (ezyang) [inline .circleci/verbatim-sources/pytorch-build-params.yml]: nit: if the parameters are only used from only one location, it's clearer if they're inlined at the single use site (you do not have to do anything about this comment)

**Review by ezyang — APPROVED** [review]
okey dokey

File: .circleci/config.yml (hunk 1 above)
Comment (xta0) [reply .circleci/config.yml]: I also added the `libtool` at the end. I remember  I ran into some error when running `brew untap caskroom/homebrew-cask`.

**Review by kostmo — APPROVED** [review]

**Pull request closed without merging**